File: QuickSort.js
let a=[10,5,4,2,20];
let  lowerBound = 0;
 let upperBound = a.length-1;
 console.log("This is an Unsorted Array "+a);


function partition(lowerBound,upperBound) {
    let p,q,pivot;
    pivot = a[lowerBound];
    p =lowerBound;
    q =upperBound;
    while(p<q){
        do {
            p++;
        } while (a[p]<=pivot && p<=upperBound);

     while (pivot< a[q]){
        q--;
    };

        if(p<q){
            let temp = a[p];
            a[p] =a[q];
            a[q] = temp;
        }
    }
    let temp = a[lowerBound];
    a[lowerBound] = a[q];
    a[q] = temp;
    return q;
};


function quickSortAlgorithm(lowerBound,upperBound){
    if(lowerBound<upperBound){
        let part = partition(lowerBound,upperBound);
        quickSortAlgorithm(lowerBound,part-1);
        quickSortAlgorithm(part+1,upperBound);
    }
}

quickSortAlgorithm(lowerBound,upperBound);
console.log(`This is Sorted Using Quick Sort Aproach ${a}`);

File: QuickSort.c
#include<stdio.h>
#include<stdlib.h>

void printingList(int arr[],int sizeOfArr){
    for(int i =0;i<sizeOfArr;i++){
        printf("%d ",arr[i]);
    }
}

int partition(int arr[],int left,int right){
    int pivot = arr[left];
    int p =left;
    int q = right;

    while(p<q){
        do{
            p++;
        }while(pivot>arr[p]);
        do{
            q--;
        }while(pivot<arr[q]);

        if(p<q){
           int temp = arr[p];
           arr[p] =arr[q];
           arr[q] =temp;
        }
    }
    int temp = arr[left];
    arr[left] = arr[q];
    arr[q] = temp;
    return q;
};

void quickSort(int arr[],int left,int right){
    if(left <right){
    int part = partition(arr,left,right);
    quickSort(arr,left,part-1);
    quickSort(arr,part+1,right);
    }
}

void main(){
    int arr [] ={41,35,11,44,75};
    int sizeOfArr = sizeof(arr)/sizeof(arr[0]);

    printf("Unsorted Array:- ");
    printingList(arr,sizeOfArr);
    // printf("%d",sizeOfArr);
    quickSort(arr,0,sizeOfArr-1);
    
    printf("\nSorted Array:- ");
    printingList(arr,sizeOfArr);
}